utils/cloudevent.go

```diff
@@ -85,8 +85,16 @@ func DoCloudEventOnce(handler Handler, ctx context.Context, in io.Reader, out io
 		ctx, cancel := CtxWithDeadline(ctx, ceIn.Extensions.Deadline)
 		defer cancel()
 
-		handler.Serve(ctx, strings.NewReader(ceIn.Data.(string)), &resp)
+		if ceIn.ContentType == "application/json" {
+			buf.Reset()
+			data := ceIn.Data.(map[string]interface{})
+			err = json.NewEncoder(buf).Encode(data)
+			handler.Serve(ctx, buf, &resp)
```

Comment:
this won't work, exactly. e.g. in the case the handler doesn't consume all the input before giving any output, which shouldn't happen, but could -- say for a function that ignores the input and just writes output (this seems quite likely to happen). needs something like:

```
in := strings.NewReader(buf.String()) // string is immutable, we need a copy
buf.Reset()
handler.Serve(ctx, in, &resp)
```

where the buf.Reset() is located in the current code (right after the `if`) is gratuitous, we're already clearing the buffer at the head of this method, so that can be removed. 

anyway this is really unfortunate and hopefully we can have more cloud event native fdks soon